src/testdrive/src/action/version_check.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use anyhow::{bail, Context};
+use tokio_postgres::types::Type;
+
+use crate::action::{ControlFlow, State};
+
+pub async fn run_version_check(
+    min_version: i32,
+    max_version: i32,
+    state: &mut State,
+) -> Result<ControlFlow, anyhow::Error> {
+    let query = "SELECT mz_version_num()";
+    let stmt = state
+        .pgclient
+        .prepare(query)
+        .await
+        .context("failed to prepare version-check query")?;
+    if stmt.columns().len() != 1 || *stmt.columns()[0].type_() != Type::INT4 {
+        bail!(
+            "version-check query must return exactly one int column, but is {}",
```

Comment:
Nit: May be misleading if the first value is INT but more than one column is returned. But this will probably never be the case...

Reply:
For me "must return exactly one int column" implies "no other columns, even if one happens to be int"

Reply:
You are right